src/clj_kondo/impl/analyzer.clj

```diff
@@ -2597,15 +2597,14 @@
           (when (:k expr)
             (usages/analyze-keyword ctx expr)
             (types/add-arg-type-from-expr ctx expr))
-          (when-not (= :edn (:lang ctx))
-            (let [id (gensym)
-                  expr (assoc expr :id id)
-                  _ (analyze-usages2 ctx expr)
-                  usage (get @(:calls-by-id ctx) id)]
-              (if usage
-                (types/add-arg-type-from-usage ctx usage expr)
-                (types/add-arg-type-from-expr ctx expr))
-              nil)))
+          (let [id (gensym)
```

Comment:
Maybe we can keep the old behavior when keyword analysis isn't enabled since this will be more performant

Reply:
Makes sense, a little bit odd make a check if keywords is enabled here though

Reply:
Why?

Reply:
Not a big deal, it's just that seems a little bit coupled.
I can do that

Reply:
The goal is to not do extra work when you're not using keyword analysis. How this is accomplished is secondary to me.